railties/lib/rails/application/finisher.rb

```diff
@@ -138,6 +138,9 @@ def self.complete(_state)
           # some sort of reloaders dependency support, to be added.
           require_unload_lock!
           reloader.execute
+          if app.config.eager_load
+            app.routes.eager_load!
```

Comment:
It can be in another one. But it has to be called after `reloader.execute` so that routes are loaded.

I figured since this is just a byproduct of loading routes, it would fit nicely there.